core/src/main/java/org/apache/iceberg/ManifestEntriesTable.java

```diff
@@ -129,7 +129,7 @@ protected long targetSplitSize(TableOperations ops) {
     public CloseableIterable<StructLike> rows() {
       return CloseableIterable.transform(
           ManifestFiles.read(manifest, io).project(fileSchema).allEntries(),
-          file -> (GenericManifestEntry) file);
+          GenericDataFile.AsManifestEntry.class::cast);
```

Comment:
This keeps the old schema because we are using `ManifestEntry.getSchema(partitionType)` in `AsManifestEntry`, right?

Reply:
Yes. This table continues to use ManifestEntry as a view of the data.